src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,180 @@
+from typing import Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+
+from unstract.sdk.exceptions import FileOperationError
+from unstract.sdk.file_storage.constants import Common, FileSeekPosition
+from unstract.sdk.file_storage.fs_interface import FileStorageInterface
+from unstract.sdk.file_storage.helper import FileStorageHelper
+
+
+class FileStorage(FileStorageInterface):
+    # This class integrates fsspec library for file operations
+
+    fs: fsspec  # fsspec file system handle
+
+    def __init__(self, type, credentials: dict() = {}):
```

Comment:
@gaya3-zipstack where will these credentials and type be passed from in our final implementation.